### ISSUE TITLE
BE - Node Mongoose Warning `errors` is a reserved schema pathname and may break some functionality

### ISSUE THREAD

**Issue by mertssmnoglu** [bluewave-labs/checkmate#1539]
**Describe the bug**

Checkmate uses [Capture](https://github.com/bluewave-labs/capture) as an Infrastructure monitoring agent. The Capture API Response includes `errors` field to correlate the data that cannot be retrieved. This responses are stored in MongoDB with the same key names, but since the `errors' object is a reserved schema pathname, Mongoose throws a warning as follows.

```shell
(node:544409) [MONGOOSE] Warning: `errors` is a reserved schema pathname and may break some functionality. You are allowed to use it, but use at your own risk. To disable this warning pass `suppressReservedKeysWarning` as a schema option.
```

**To Reproduce**
1. Go to **Server/**
2. Install dependencies `npm install`
3. Run `npm run dev`
4. Scroll up to head of the logs
5. See the warning starts with `(node:544409) [MONGOOSE] Warning`

**Expected behavior**
Seeing the log messages clearly in the terminal without any errors or warnings.

**Additional context**
For more information about reserved keys you can visit [MongooseDocs#Schema.reserved](https://mongoosejs.com/docs/api/schema.html#Schema.reserved)

**Possible Actions**

1. Changing the DB CRUD methods
    It will be more useful and safe solution for everyone.
2. Changing the API Schema of the Capture
    A new version of Capture might be required for the users.  
3. Disabling the warning.
    To disable this warning pass `suppressReservedKeysWarning` as a schema option.

**Comment by ajhollid** [comment]
@mertssmnoglu can you elaborate on possible action 1?  What are you thinking here?

**Comment by mertssmnoglu** [comment]
Hey @ajhollid, I'm thinking about renaming the `errors` key in the `Hardware Check Model` because it's reserved on the mongoose.

https://github.com/bluewave-labs/checkmate/blob/0e443e4088ffe2f1f140aaa7717d7a45889bc41f/Server/db/models/HardwareCheck.js#L58-L61

My Options
```json
[
  "error", "hardwareErrors", "hardwareError", "infrastructureErrors",
  "infraErrors", "hwErr", "errorsObj", "errObj", "err"
]
```

---

Spread syntax used here, because all of the keys are same as the `Hardware Check Model`
If we decide to rename the key, we will no longer be able to use spread syntax on the `Hardware Check Module`

https://github.com/bluewave-labs/checkmate/blob/0e443e4088ffe2f1f140aaa7717d7a45889bc41f/Server/db/mongo/modules/hardwareCheckModule.js#L35-L37

New code block will be like that

```js
const hardwareCheck = await new HardwareCheck({
	//comes from "Base Check Model"
	monitorId: hardwareCheckData.monitorId,
	status: hardwareCheckData.status,
	responseTime: hardwareCheckData.responseTime,
	statusCode: hardwareCheckData.statusCode,
	message: hardwareCheckData.message,
	expiry: hardwareCheckData.expiry,

	cpu: hardwareCheckData.cpu,
	memory: hardwareCheckData.memory,
	disk: hardwareCheckData.disk,
	host: hardwareCheckData.host,	
	
	SOME_NEW_ERRORS_KEY: hardwareCheckData.errors
}).save();
```

Yes, spread operator is very clean and useful but I think we should rename the `errors` with something different. Because of [this](https://mongoosejs.com/docs/api/schema.html#Schema.reserved). I'm not sure about how big is the "Breaking Plugin Functionality" issue.

**Comment by ajhollid** [comment]
@mertssmnoglu sure we can do whichever you prefer, it's not a big deal either way.

Why don't you make the call and we'll adjust our backend to fit?

**Comment by Stitch10925** [comment]
"faults"?
